torch/csrc/jit/export.cpp

```diff
@@ -745,22 +830,39 @@ void ScriptModuleSerializer::convertModule(
     convertParameter(elem, param_def, /*is_buffer=*/false);
   }
 
+  // If __getstate__ and __setstate__ methods are provided, use those for
+  // serializing instead of serializing the attributes directly
+  bool user_provided_serialization = moduleHasValidGetSetState(module);
+  if (user_provided_serialization) {
+    // Run the '__getstate__' method on the module and store the result
+    pickled_ivalues_.emplace_back(moduleGetState(module));
+    module_def->set_id(pickled_ivalues_.size() - 1);
+  }
   for (const auto& attribute : module.get_attributes()) {
     // Add attribute to ModuleDef
     torch::AttributeDef* attribute_def = module_def->add_attributes();
     attribute_def->set_name(attribute.name());
     attribute_def->set_type(attribute.type()->python_str());
 
-    attribute_table_.push_back(attribute.value());
-    attribute_def->set_id(attribute_table_.size() - 1);
+    if (!user_provided_serialization) {
```

Comment:
We should not write out parameters as well. This features exists because there is a non-serializable tensor format being used for a parameter.